taxjar_salestax/models/account_invoice_line.py

```diff
@@ -31,8 +31,9 @@ def _get_taxjar_lines(self, config, commit=False):
         description are present in each line.
         """
         items = []
+        only_shipping = all(l.product_id in config.shipping_product_ids for l in self)
         for line in self:
-            if line.product_id in config.shipping_product_ids:
+            if not only_shipping and line.product_id in config.shipping_product_ids:
```

Comment:
As discussed in our call, we can use the Odoo shipping order lines as line items, but you'd have to set the amount to zero so that the total invoice amount is still equal to the shipping amount in the payload plus the amounts from the items in the payload.

Reply:
Thanks again Stefan, this part is accounted for now. ^^
I believe the logic should be fine now, I'll just try to test a few scenarios in the sandbox environment of TaxJar.